src/asset-swapper/utils/gas_price_utils.ts

```diff
@@ -41,16 +42,26 @@ export class ProtocolFeeUtils {
         return instance;
     }
 
-    /** @returns gas price (in wei) */
-    public async getGasPriceEstimationOrThrowAsync(shouldHardRefresh?: boolean): Promise<BigNumber> {
-        if (this._gasPriceEstimation.eq(constants.ZERO_AMOUNT)) {
-            return this._getGasPriceFromGasStationOrThrowAsync();
+    public async getGasPriceEstimationOrDefault(defaultGasPrices: GasPrices): Promise<GasPrices> {
+        if (this._gasPriceEstimation === undefined){
```

Comment:
nit: if `_updateGasPriceFromOracleOrThrow` fails for some reason, it will throw an error rather than returning the default. 

I think it's fine to return the default if `this._gasPriceEstimation` is undefined. 
If we are calling `_updateGasPriceFromOracleOrThrow` then I think we should swallow the error and return the default.